homeassistant/components/light/hue.py

```diff
@@ -56,25 +54,61 @@
     cur_lights = {}
     cur_groups = {}
 
-    await async_update_lights(
-        hass, bridge, cur_lights, cur_groups, async_add_devices)
-
+    # Hue updates all lights via a single API call.
+    #
+    # If we call a service to update 2 lights, we only want the API to be
+    # called once.
+    #
+    # The throttle decorator will return right away if a call is currently
+    # in progress. This means that if we are updating 2 lights, the first one
+    # is in the update method, the second one will skip it and assume the
+    # update went through and updates it's data, not good!
+    #
+    # The @async_join_concurrent will make sure that all lights will wait till
+    # the update call is done before writing their data to the state machine.
+    #
+    # One thing to note is that a call to `update_bridge` will call
+    # `async_schedule_update_ha_state` on all lights and groups, including the
+    # ones that are about to be written and triggered the call to
+    # `update_bridge` to begin with! There are 2 scenarios why we do this:
+    #
+    #  1. The bridge is no longer available. We trigger an update on all lights
+    #     and groups to make sure they mark themselves as unavailable.
+    #  2. If we are updating 1 light with a service call, we also want to
+    #     capture changes that occured to other lights (either via outside
+    #     control or because the 1 light was actually a group in Hue).
+    #
+    # An alternative approach would be to disable automatic polling by Home
+    # Assistant and take control ourselves. This works great for polling as now
+    # we trigger from 1 time update an update to all entities. However it gets
+    # tricky from inside async_turn_on and async_turn_off.
+    #
+    # If automatic polling is enabled, Home Assistant will call the entity
+    # update method after it is done calling all the services. This means that
+    # when we update, we know all commands have been processed. If we trigger
+    # the update from inside async_turn_on, the update will not be capture the
+    # changes to the second entity until the next polling update.
```

Comment:
Clarify why the update will not capture the changes to the second entity (reason: Throttle).